### DISCUSSION TITLE
Some suggestions

### DISCUSSION THREAD

**Discussion by igkiou** [cvpr-org/author-kit#36] [Ideas]
Hi,

I'd like to make a few suggestions for the template. Happy to make a pull request (or separate ones per each suggestion) if there is general agreement for the changes.

1. Configure `cleveref` so that it uses abbreviations *consistently*–either always or never abbreviate mid-sentence references. Right now, `cleveref` abbreviates "Sec." and "Tab.", but it doesn't abbreviate "Figure", "Algorithm", or "Equation". (References at the start of a sentence are already always unabbreviated, as they should be.) 

2. Include `microtype` in the packages in `cvpr.sty`, which helps improve kerning and spacing. Personally I prefer to use it with the default settings, but opinions differ.

3. Include the "inline" option when loading `enumitem` in `cvpr.sty`. This makes it possible to use the `enumerate*` environment to easily and consistently produce inline enumerated lists, e.g., "We achieve this by 1. doing this; and 2. doing that."

4. Configure `hyperref` to use `cvprblue` for all links, by changing the current option `citecolor=cvprblue` to `allcolors=cvprblue`. Under the current settings, we end up with different colors for citations versus references to sections, equations, etc., versus URLs. The resulting rainbow of colors throughout the text ends up being very distracting.

5. Configure `hyperref` to automatically produce correct PDF metadata in the camera-ready PDF. This is already implemented in, e.g., the SIGGRAPH LaTeX template, meaning that final PDFs for SIGGRAPH papers automatically have correct titles, authors, etc.

**Comment by taiya** [comment]
1. this is already controllable from the user via \Cref vs. \cref, no?
2. problem is microtype adds a fair bit of overhead to compile. I have used it consistently, but usually enable it at the very very end, as it allows you to squeeze a bit of extra space. We could add a commented-out line, or to the camera ready version, but I am hesitant setting it to default is a good idea; @cr333 ?
3. seems like a good idea; send a pull request?
4. as far as I remember, sections vs. citations always used a different color. Do I remember incorrectly? This was from a CVPR 2020 paper, for example: https://capture.dropbox.com/0DmIXJ4picOjDTOz
5. if the metadata is wrong, this seems like a great suggestion. Do you want to create a separate issue, and perhaps a PR with a proposed fix? Seems you know about this more than we do.

p.s. this entire post seemed a better fit for the discussion tab :)

**Reply by igkiou** [reply]
Discussion vs issue: Absolutely, thanks for converting it!

1. ~~It is controllable, but I was suggesting changing the *default* behavior of the template. Currently, `cvpr.sty` has `\cref` commands [link](https://github.com/cvpr-org/author-kit/blob/47dfc5c0eee09dcceac65e069a5180fb871008ef/cvpr.sty#L616) only for "Sec." and "Tab.", while leaving everything else unabbreviated. As a result, most CVPR authors who will go with the default behavior will end up with papers that have a mix and match of abbreviated and unabbreviated references. What I was suggesting is changing the default behavior to either abbreviate everything or nothing.~~ Nevermind, see below.

2. Fair point. My personal view is that the benefits outweigh the increase in compilation time, so I would go with having it on by default. But obviously that's very subjective, so I'll let you all decide what you think is best. [I'd make a bet that if more authors knew it helps save space, they'd all choose to enable it just for that!]

3. Will do!

4. Yeah, it's not a new behavior. What I was suggesting is that, now that there is a designated color for hyperlinked references (great choice of color by the way), the template should change to use it for all hyperlinks. At least in my view, it makes hyperlinks a lot less distracting without causing any loss of semantic information (it's pretty obvious from context whether a hyperlink is a citation, section, equation, etc.)

5. Will do!

**Reply by igkiou** [reply]
Actually, regarding 1: I tested out compiling on a different system, and it looks like the default template *does* consistently abbreviate everything---and this behavior is indeed in agreement with the documentation for `cleveref`. 

So the inconsistent abbreviation seems like it's a local issue on my end. Sorry for the confusion.

**Reply by taiya** [reply]
Sounds good, keep in mind you can also tweak the class file to adjust the default behavior.
I do that consistently in most my submissions.

If I recall correctly, it's at the very end of it.

**Reply by igkiou** [reply]
Of course, I do [likewise](https://github.com/cmu-ci-lab/writing). 

I'll go ahead and put together separate pull requests for each of the other items (including `microtype`, to help better localize discussion) in the next few days.

**Comment by cr333** [comment]
Thanks so much for these suggestions, Ioannis!

1. I would argue for always spelling out the names in references, as it makes text more readable due to less mental overhead for decoding "Fig.", "Tab." etc., while not really require that much more space – usually just 2 to 3 extra letters that we should be able to afford. But this is a stylistic choice without a clear consensus, so at least the style should be consistent within each paper.
2. I'm a big fan of microtype as it improves typography while making text a little bit bit more compact.
3. Sure.
4. This seems like a good change to me, as it makes link colours consistent in papers. The CVPR blue is also easier on the eyes than the saturated red used used for internal references, or the stark pink of URLs.
5. A PR would be much appreciated!

**Reply by igkiou** [reply]
1. Agreed on all fronts. I actually also go for fully spelt-out references in my own submissions, but I know that this is a very subjective choice. My suggestion was intended only to ensure that the default behavior is consistent, rather than argue for one versus the other. I'd be all for changing the default behavior to be fully spelt-out references, but that probably merits some further discussion.

2--4. Great, I already submitted PRs for these three simple changes.

5. I started working on one, though it'll likely come after the ECCV deadline.
